### DISCUSSION TITLE
Azure storage HTTP/SOCKS proxy support

### DISCUSSION THREAD

**Discussion by Codesleuth** [delta-io/delta-rs#2943] [Q&A] [answered]
I'm trying to access an Azure Storage Account with `DeltaTable` in python, while specifying `HTTPS_PROXY`, but it seems delta-rs doesn't respect the environment variable. Is support for proxies still available in the library? I can see issues which show it has been added in the past (for S3 specifically) but that was before the code was vastly refactored.

I guess my question is: how can I use a proxy for access to Azure Storage Accounts?

**Comment by Codesleuth** [comment, marked as answer]
Quick update, although still not working. This now gives me a different error that appears to change the behaviour to use the proxy:

```python
delta_table = DeltaTable(
    table_uri=table_path,
    storage_options={
        "bearer_token": token.token,
        "proxy_url": environ.get("HTTPS_PROXY", None),
    },
)
```

These are the errors:
```
# HTTPS_PROXY=socks5://localhost:1080
Exception: OSError: Generic HTTP client error: builder error

# HTTPS_PROXY=http://localhost:8080
Exception: OSError: Generic MicrosoftAzure error: Error after 10 retries in 3.536299418s, max_retries:10, retry_timeout:180s, source:error sending request for url ([...])
```

So it appears a SOCKS server just doesn't work, and a web proxy is failing for another reason (possibly my end).

**Reply by Codesleuth** [reply]
Can confirm that the `"proxy_url"` setting with a web (http) proxy (not SOCKS) is working. However I think it's not a bad idea to have proxy support inherit the environment variable for it as most would expect.

**Reply by ion-elgreco** [reply]
@Codesleuth that's something you should bring up in the arrow-rs repo, there is the crate we use for objectstore interaction maintained: https://github.com/apache/arrow-rs
